### PR TITLE
Updated WorldSocket::HandleAuthSession

### PR DESCRIPTION
**Pull request by sanctum32**
In this pr I did these changes:
* Removed invalid variable missmatch
* Reduced variable scopes
* Simplified sha digest checks
* Initialized previously non initialized variables

Tests performed:
* Logged to world
* Did relogs
* Tried several accounts

Tested on:
windows 10 64bit
openssl v 3.3.0
msvc 2022
mysql 8.0.37

- - -
This change is [Reviewable](https://reviewable.io/reviews/mangoszero/server/197)

### CODE REVIEW

**Comment by Meltie2013** [comment]
@sanctum32 
```
D:\a\server\server\src\shared\Utilities\Util.h(137,63): error C2039: 'ptr_fun': is not a member of 'std'
```

**Comment by sanctum32** [comment]
> @sanctum32
> 
> ```
> D:\a\server\server\src\shared\Utilities\Util.h(137,63): error C2039: 'ptr_fun': is not a member of 'std'
> ```

this pr is not related to this error, seems C++17 was enabled in SD3 and that triggered chain errors

File: src/game/Server/WorldSocket.cpp [hunk 1]
```diff
@@ -766,33 +752,30 @@ int WorldSocket::HandleAuthSession(WorldPacket& recvPacket)
...
     sha.UpdateData(account);
     sha.UpdateData((uint8*) & t, 4);
     sha.UpdateData((uint8*) & clientSeed, 4);
     sha.UpdateData((uint8*) & seed, 4);
-    sha.UpdateBigNumbers(&K, NULL);
+    sha.UpdateBigNumbers(&K, nullptr);
     sha.Finalize();
 
-    if (memcmp(sha.GetDigest(), digest, 20))
+    if (sha.GetDigest() != digest)
```
Comment (H0zen) [inline src/game/Server/WorldSocket.cpp]: This is not ok! Comparison of 2 arrays

Comment (sanctum32) [reply src/game/Server/WorldSocket.cpp]: right, for this std::array has to be involved... ech, bits of outdated content... will correct this. This comparition is valid between std::array.

**Pull request merged**